impl/src/main/java/org/jboss/weld/util/Proxies.java

```diff
@@ -56,32 +59,61 @@
 
         private final List<Class<?>> interfaces;
         private final List<Class<?>> classes;
+        private final Map<String, String> classToPackageMap;
 
         private TypeInfo(Set<? extends Type> types) {
-            Comparator<Class<?>> classComparator = Comparator.comparing(Class::getName);
             List<Class<?>> foundInterfaces = new ArrayList<>();
             List<Class<?>> foundClasses = new ArrayList<>();
+            Map<String, String> classToPackage = new HashMap<>();
 
-            types.stream().forEach(type -> add(type, foundInterfaces, foundClasses));
+            types.stream().forEach(type -> add(type, foundInterfaces, foundClasses, classToPackage));
 
             // sort both collections and create unmodifiable lists
-            Collections.sort(foundClasses, classComparator);
-            Collections.sort(foundInterfaces, classComparator);
+            Collections.sort(foundClasses, Comparator.comparing(Class::getName));
+            Collections.sort(foundInterfaces, (o1, o2) -> {
+                String javaPrefix = "java.";
+                String javaxPrefix = "javax.";
+                String firstClassName = o1.getName();
+                String secondClassName = o2.getName();
+                // if no class starts with java.* or javax.* or if both start with it, perform standard comparison
+                // of only one starts with this prefix, it goes later
+                boolean firstClassHasJavaPrefix = firstClassName.startsWith(javaPrefix) || firstClassName.startsWith(javaxPrefix);
+                boolean secondClassHasJavaPrefix = secondClassName.startsWith(javaPrefix) || secondClassName.startsWith(javaxPrefix);
+                if (firstClassHasJavaPrefix) {
+                    if (secondClassHasJavaPrefix) {
+                        // both classes prefixed
+                        return firstClassName.compareTo(secondClassName);
+                    } else {
+                        // first class prefixed, second class not
+                        return 1;
+                    }
+                } else {
+                    if (secondClassHasJavaPrefix) {
+                        // first class is not prefixed, second class is
+                        return -1;
+                    } else {
+                        // neither class is prefixed
+                        return firstClassName.compareTo(secondClassName);
+                    }
+                }
+            });
             this.interfaces = Collections.unmodifiableList(foundInterfaces);
             this.classes = Collections.unmodifiableList(foundClasses);
```

Comment:
I know that it's not related to this PR but it would not hurt to use ``ImmutableList``s here as well...